kotlin/export-tiles/src/main/java/com/example/exporttiles/MainActivity.kt

```diff
@@ -0,0 +1,229 @@
+          mapView.getWidth() - 150,
+          mapView.getHeight() - 250
+        )
+        // convert screen points to map points
+        val minPoint: Point? = mapView?.screenToLocation(minScreenPoint)
+        val maxPoint: Point? = mapView?.screenToLocation(maxScreenPoint)
+        // use the points to define and return an envelope
+        downloadArea.geometry = Envelope(minPoint, maxPoint)
+      }
+    }
+
+    // create up a temporary directory in the app's cache for saving exported tiles
+    val exportTilesDirectory = File(cacheDir, getString(R.string.tile_cache_folder))
+
+    // when the button is clicked, export the tiles to a temporary file
+    exportTilesButton.setOnClickListener {
+      val exportTileCacheTask = ExportTileCacheTask(tiledLayer.uri)
+      val parametersFuture: ListenableFuture<ExportTileCacheParameters> =
+        exportTileCacheTask.createDefaultExportTileCacheParametersAsync(
+          downloadArea.geometry,
+          mapView.mapScale,
+          tiledLayer.maxScale
+        )
+
+      parametersFuture.addDoneListener {
+        val parameters: ExportTileCacheParameters = parametersFuture.get()
+        // export tiles to temporary cache on device
+        val exportTileCacheJob =
+          exportTileCacheTask.exportTileCache(parameters, exportTilesDirectory.path + "file.tpk")
+        exportTileCacheJob.start()
+        // create and show a progress dialog to show export tile progress
+        val dialog = createProgressDialog(exportTileCacheJob)
+        dialog.show()
+
+        // show progress of the export tile cache job on the progress bar
+        exportTileCacheJob.addProgressChangedListener {
+          dialog.progressBar.progress = exportTileCacheJob.progress
+        }
+
+        exportTileCacheJob.addJobDoneListener {
+          dialog.dismiss()
+          if (exportTileCacheJob.status == Job.Status.SUCCEEDED) {
+            val exportedTileCacheResult: TileCache = exportTileCacheJob.result
+            showMapPreview(exportedTileCacheResult)
+          } else {
+            Log.e(TAG, exportTileCacheJob.error.additionalMessage)
+            Toast.makeText(
+              this, exportTileCacheJob.error.additionalMessage, Toast.LENGTH_LONG
+            ).show()
+          }
+        }
+      }
+    }
+
+    // close the preview window
+    closeButton.setOnClickListener { clearPreview() }
+    clearPreview()
+  }
+
+  /**
+   * Show tile cache preview window including containing the exported tils.
+   *
+   * @param result takes the TileCache from the ExportTileCacheJob
+   */
+  private fun showMapPreview(result: TileCache) {
+    val newTiledLayer = ArcGISTiledLayer(result)
+    val previewMap = ArcGISMap(Basemap(newTiledLayer))
+
+    previewMapView.apply {
+      map = previewMap
+      setViewpoint(mapView.getCurrentViewpoint(Viewpoint.Type.CENTER_AND_SCALE))
+      visibility = View.VISIBLE
+      getChildAt(0).visibility = View.VISIBLE
+    }
+    mapPreviewLayout.bringToFront()
```

Comment:
I think its probably confusing using z-ordering here. When I wrote the java version of this I didn't have a good understanding of Android UI. Can we just use `visibility=View.VISIBLE` here and invisible elsewhere?